### ISSUE TITLE
Navigation not working on Android 7.1.2

### ISSUE THREAD

**Issue by surferjoeit** [junjunguo/PocketMaps#138]
Hello, 3.1 version downloaded from f-droid doen't work properly on an android 7.1.2 device.
I manage to set the starting point with GPS position, but when routing starts the position is stuck at the starting point, as if GPS signal is not received anymore.
Thanks!

**Comment by Starcommander** [comment]
Hello @surferjoeit !
Thanks for the message.
Can you try to switch off/on the kalman-filter.
Maybe it has to do with that new feature.
You can find it on navigation-settings "Smoothen movement"

Best regards,
Paul

**Comment by Starcommander** [comment]
Of course, you have GPS switched on, right?
Can you still see the GPS-Symbol on the taskbar while no signal is received?
And which kind of gps does the toast message show, when re-opening Pocketmaps?
There are "kalman" "gps" or "passive"

**Comment by Starcommander** [comment]
49cc6df
Crazy, now I have a hint, why it was not possible to get GPS Signal last week.

**Comment by Starcommander** [comment]
@surferjoeit Maybe you can re-test with this changes?
I am interested in:
- GPS is working for navigation?
- GPS with kalman-filter (Smooth) is working for navigation?

Important: This APK may remove your maps/favourites/trackingrecords, because you have to remove PocketMaps-app first.
778c339

Thanks in advance
Paul

**Comment by surferjoeit** [comment]
> 
> 
> Of course, you have GPS switched on, right?
> Can you still see the GPS-Symbol on the taskbar while no signal is received?
> And which kind of gps does the toast message show, when re-opening Pocketmaps?
> There are "kalman" "gps" or "passive"

ahahahahah, GPS was switched on, of course, that was my first verification.....
anyway, this morning i made some more deep testing going to work... here is what happens:

- with kalman-filter off routing doesn't work, as previously described, and location provider message says: gps. This morning on the first attempt the app crashed, on the second one the position was stuck at starting point as previously described
- with kalman-filter on routing works, even if it (always) asks to switch on GPS even if it's already active....

hope this helps you. Do you confirm to try the above attached app or do i have to wait for some other modifications?
Some other random comments:

- translation of kalman-filter toggle should be improved: now italian translation is "legare il movimento", quite confusing.. would it be possible to modify it in "movimento fluido"?
- speech synthesis is always in english. Not a big issue, but it would be nice to have it according to device language
- the app keeps on creating a "pocketmaps" folder in the download folder even if map location has been moved to another position. Again not a big issue...

Many thanks!
Francesco

**Comment by Starcommander** [comment]
@surferjoeit 
Of course, modify translation is no problem.
You can find all translations in the file **PocketMaps/app/src/main/res/values-it/strings.xml**
Maybe you find further phrases to improve.

If you have a crash of Pocketmaps, then a logcat-output is really helpful in most times.
And yes, the attached test-app should solve the issue.
Also solved: "asks to switch on GPS"

If you also can confirm that this app solves the issue, then I will release a new version as fast as possible to playstore and fdroid, because this issue is really severe.

Thanks and best regards,
Paul

**Comment by surferjoeit** [comment]
Hello,
Unfortunately the issue is not solved.
Uninstalled previous version, installed the new one, same issue...
I'll check strings, no problem
Thanks

**Comment by Starcommander** [comment]
@surferjoeit 
Can you please explain a little bit more?
Can you see the GPS-Symbol on the notification bar?
What exactly happens, when you switch on/off the kalman-filter (smoothen m.)?
Does the app find your location?
Does the location sign move, when you change your position?
Does the app still ask for enabling GPS, when already activated?
What does the toast-message show as location-provider?

**Comment by surferjoeit** [comment]
Answers between tour questions.

> @surferjoeit
> Can you please explain a little bit more?
> Can you see the GPS-Symbol on the notification bar?

Only when kalman is on

> What exactly happens, when you switch on/off the kalman-filter (smoothen m.)?

When i switch on kalman, firstly the app asks for GPS activation, even if it's already activated, and routing starts working and car position starts changing. If i switch kalman off position freezes.

> Does the app find your location?

When kalman is off only when i define starting position by pressing the target i con. When kalman is on also during routing.

> Does the location sign move, when you change your position?

Only with kalman on

> Does the app still ask for enabling GPS, when already activated?

Only the first time, at app launch and only if kalman is on. Or whenever i switch kalman from off to on.

> What does the toast-message show as location-provider?

With kalman on: kalman
With kalman off: GPS (if activated), but with no GPS icon in the notification bar

Many thanks

**Comment by Starcommander** [comment]
Ok, thanks for your clarification.

**Comment by surferjoeit** [comment]
Hope it's enough to help you!

**Comment by Starcommander** [comment]
@surferjoeit Maybe you can retest the newest pre-release?
https://github.com/junjunguo/PocketMaps/releases/tag/3.3-pre1
- Check if app-version is really 3.3
- Check if kalman and gps are working for navigation.
- Check if Activate-GPS dialog appears again.

Thanks in advance
Paul

**Comment by surferjoeit** [comment]
@Starcommander some good and bad news...
let's start with good ones:
1. app version is 3.3
2. activate gps dialog doesn't appear
3. kalman is working fine, gps works only after having switched off kalman (but i'll retest it later today, i made a quick test this morning but...
and now a bad one comes...
4. after a short trip where it worked as described on item 3 (gps at the very first start didn't work), after the devices screen turned off, the app closed.. and when i reopened it it asked to download the map because the installed one was not compatible... and the installed map disappeared. i've just re-downloaded the map and i'll re-test the app later today.
Thanks for your support!

**Comment by surferjoeit** [comment]
I made some other testing and confirm the above mentioned  behaviour.
The main issue is that it seems that every time i close and reopen the app it requires to download the map again and the downloaded map disappear (even if files are still in the proper folder).

**Comment by Starcommander** [comment]
@surferjoeit 
How exactly do you "close and reopen the app"?
I never had this issue.
And is everytime the message "not compatible" showing or does the map just disapear?
Maybe you can create a logcat-output directly after the issue, would be very helpful!

Thanks in advance,
Paul

**Comment by surferjoeit** [comment]
With close i mean when i press the home button, after routing end.
The not compatibile message comes first, then, if i "close" and reopen the app, the map desappears.
Map files are still in the proper folder, anyway.
How to create a logcat? My device is not rooted, is it possibile anyway?
Thanks


Il 24 gennaio 2020 19:44:58 CET, Starcommander <notifications@github.com> ha scritto:
>@surferjoeit 
>How exactly do you "close and reopen the app"?
>I never had this issue.
>And is everytime the message "not compatible" showing or does the map just disapear?
>Maybe you can create a logcat-output directly after the issue, would be very helpful!
>
>Thanks in advance,
>Paul
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/junjunguo/PocketMaps/issues/138#issuecomment-578252117
-- 
Inviato dal mio dispositivo Android con K-9 Mail. Perdonate la brevità.

**Comment by Starcommander** [comment]
@surferjoeit 
No root needed.
What you need is the ADB (Android debug bridge).
- https://adbinstaller.com/

Also enable adb debugging on your device as described here
- https://developer.android.com/studio/command-line/adb

Connect your phone with PC, and execute: adb logcat
This will output all the messages that I need.

**Comment by Starcommander** [comment]
Also useful: note the time, when the issue appears.

**Comment by surferjoeit** [comment]
I'll try...
But probably you'll have to wait a couple of days.
Thanks!

Il 24 gennaio 2020 20:15:43 CET, Starcommander <notifications@github.com> ha scritto:
>@surferjoeit 
>No root needed.
>What you need is the ADB (Android debug bridge).
>- https://adbinstaller.com/
>
>Also enable adb debugging on your device as described here
>- https://developer.android.com/studio/command-line/adb
>
>Connect your phone with PC, and execute: adb logcat
>This will output all the messages that I need.
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/junjunguo/PocketMaps/issues/138#issuecomment-578262814
-- 
Inviato dal mio dispositivo Android con K-9 Mail. Perdonate la brevità.

**Comment by surferjoeit** [comment]
Hello,
i finally have a logcat, see attached file (hope no sensitive data on it...).
Navigation started aproximately at 8:33 without kalman.
At 8:36 i switched on kalman since position was stuck at starting point and position started to change
AT 8:37 i stopped navigation and pressed home button, when came back to pocketmaps and tried to reload map a message "ma not compatible" appeared
I the removed pocketmaps from recent app list and when i reopened it map was disappeared.
Thanks!
[logcat_01-27-2020_08-41-01.zip](https://github.com/junjunguo/PocketMaps/files/4116157/logcat_01-27-2020_08-41-01.zip)

**Comment by Starcommander** [comment]
@surferjoeit 
Thanks for the information.
For security reason you should not forget to disable ADB-access on your phone again.
Especially if you connect your phone on a foreign computer.

Issue with GPS is reproducible and should be be solved with dda5fe2

**Comment by Starcommander** [comment]
The issue with losing map compatiblity is more complex.
```
01-27 08:37:02.569 --> FocusLost
01-27 08:37:11.547 --> Restart
01-27 08:37:11.707 --> AppLockServiceBinder
01-27 08:37:11.730 --> FocusGet
01-27 08:37:15.030 --> Map is not compatible
01-27 08:37:15.688 --> GH Compare maps Local=1970-01 Remote=2019-11
01-27 08:37:15.689 --> Update for map is available: europe_italy_nord-ovest
```
Seems the app thinks, that your local map is from 1970.
This may have to do with the version.txt file in the maps folder.
Maybe you can check the content of the version.txt file, after the issue occurs?

Also you can try if there is a difference, when changing the maps-directory in main menu.
Or trying a different map.

Thanks again.
Best regards,
Paul

**Comment by Starcommander** [comment]
Hello @surferjoeit 
I have created a new pre-release.
- GPS bug should be solved now.
- Debug output for map-compatiblity test. --> Please create a new logcat
https://github.com/junjunguo/PocketMaps/releases/tag/3.3-pre2

Also very helpful information:
Are you doing anything special before closing the app, and the issue occurs?
Does this happen every time, or not when closing the app?
Did you download the map, or copy the folder?
The result as requested above would be very helpful, too.

Thanks, and best regards,
Paul

**Comment by surferjoeit** [comment]
Thanks.
Versione.txt has the following content:
 0.13.0_0


Il 27 gennaio 2020 18:17:50 CET, Starcommander <notifications@github.com> ha scritto:
>The issue with losing map compatiblity is more complex.
>```
>01-27 08:37:02.569 --> FocusLost
>01-27 08:37:11.547 --> Restart
>01-27 08:37:11.707 --> AppLockServiceBinder
>01-27 08:37:11.730 --> FocusGet
>01-27 08:37:15.030 --> Map is not compatible
>01-27 08:37:15.688 --> GH Compare maps Local=1970-01 Remote=2019-11
>01-27 08:37:15.689 --> Update for map is available: europe_italy_nord-ovest
>```
>Seems the app thinks, that your local map is from 1970.
>This may have to do with the version.txt file in the maps folder.
>Maybe you can check the content of the version.txt file, after the issue occurs?
>
>Also you can try if there is a difference, when changing the maps-directory in main menu.
>Or trying a different map.
>
>Thanks again.
>Best regards,
>Paul
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/junjunguo/PocketMaps/issues/138#issuecomment-578854775
-- 
Inviato dal mio dispositivo Android con K-9 Mail. Perdonate la brevità.

**Comment by surferjoeit** [comment]
Hello,
I'm happy to report that everything seems to work fine.
I attached log file.
Some answers:
No, i was not doing anything special, even if it depends on what to you consider ad special...
This happened when i closed the app. As long as the app was open, no issue.
I tried both ways, or better both downloading and copying GHz file with PC in pocketmaps download folder.
Thanks!
PS. I checked italian strings and they're ok

Il 27 gennaio 2020 18:46:27 CET, Starcommander <notifications@github.com> ha scritto:
>Hello @surferjoeit 
>I have created a new pre-release.
>- GPS bug should be solved now.
>- Debug output for map-compatiblity test. --> Please create a new logcat
>https://github.com/junjunguo/PocketMaps/releases/tag/3.3-pre2
>
>Also very helpful information:
>Are you doing anything special before closing the app, and the issue occurs?
>Does this happen every time, or not when closing the app?
>Did you download the map, or copy the folder?
>The result as requested above would be very helpful, too.
>
>Thanks, and best regards,
>Paul
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/junjunguo/PocketMaps/issues/138#issuecomment-578867913
-- 
Inviato dal mio dispositivo Android con K-9 Mail. Perdonate la brevità.

**Comment by Starcommander** [comment]
Ok, then I will release a new version now.
Thanks for testing.

Best regards,
Paul

**Comment by surferjoeit** [comment]
Hello,
i come back with the not compatible map issue.
The issue is not really fixed, i just managed to understand what happens and to avoid re-downloading map.
Here is a detailed description:

- map files are located in android/media/...... pocketmaps folder

- when i open the app for the first time, everything is fine, i can open map

- if i open the option to move map folder i can see that map is located in the proper folder

- if i close the app (with back button) and reopen it and try to open map, i get the "map not compatible" message

- if i open the option to move map folder i can see now indication that map is located in download/.... folder, even if it's not true, files are still in android/media/... folder

- if i chose then to move map from download/... folder to android/media/... folder i see my installed map quickly disappear and reappear and then it works (until i close the app....)

Please find attached a log

**Comment by Starcommander** [comment]
@surferjoeit 
Please list your files and folders as follows:
- adb ls /storage/emulated/0/Download/pocketmaps/maps/europe_italy_nord-ovest-gh/
- adb ls /storage/emulated/0/Download/pocketmaps/maps/
- adb ls /storage/emulated/0/Download/pocketmaps/
- adb ls /storage/emulated/0/Download/

Also I would like to know the content of your android/media/ path.
You can see the absolute path on the change-maps-folder-dialog.

Best regards,
Paul

**Comment by surferjoeit** [comment]
Hello,
/storage/emulated/0/Download/pocketmaps/maps/europe_italy_nord-ovest-gh/ doesn't exists

Folder content:

C:\Program Files (x86)\platform-tools>adb ls /storage/emulated/0/Download/pocketmaps/maps/

000041f9 00001000 5e328708 .

000041f9 00001000 5e32871d ..


C:\Program Files (x86)\platform-tools>adb ls /storage/emulated/0/Download/pocketmaps/

000041f9 00001000 5e32871d .

000041f9 00002000 5e32c551 ..

000041f9 00001000 5e328708 maps

000041f9 00001000 5e32871d downloads


C:\Program Files (x86)\platform-tools>adb ls /storage/emulated/0/Download/

000041f9 00002000 5e32c551 .

000041f9 00001000 5e32d901 ..

000081b0 00405089 5e32c551 OpenTopoMap_1.9.1.apk

000041f9 00001000 5e32871d pocketmaps


C:\Program Files (x86)\platform-tools>adb ls /storage/emulated/0/Download/pocketmaps/maps/europe_italy_nord-ovest-gh/


C:\Program Files (x86)\platform-tools>adb ls /storage/emulated/0/Android/media/com.junjunguo.pocketmaps

000041f9 00001000 5e32868e .

000041f9 00001000 5e32868a ..

000041f9 00001000 5e3286eb pocketmaps


C:\Program Files (x86)\platform-tools>adb ls /storage/emulated/0/Android/media/com.junjunguo.pocketmaps/pocketmaps

000041f9 00001000 5e3286eb .

000041f9 00001000 5e32868e ..

000041f9 00001000 5e328150 maps

000041f9 00001000 5e328161 downloads

000081b0 0000006c 5e328161 Favourites.properties


C:\Program Files (x86)\platform-tools>adb ls /storage/emulated/0/Android/media/com.junjunguo.pocketmaps/pocketmaps/downloads

000041f9 00001000 5e328161 .

000041f9 00001000 5e3286eb ..


C:\Program Files (x86)\platform-tools>


C:\Program Files (x86)\platform-tools>adb ls /storage/emulated/0/Android/media/com.junjunguo.pocketmaps/pocketmaps/maps

000041f9 00001000 5e328150 .

000041f9 00001000 5e3286eb ..

000041f9 00001000 5e32870d europe_italy_nord-ovest-gh


C:\Program Files (x86)\platform-tools>adb ls /storage/emulated/0/Android/media/com.junjunguo.pocketmaps/pocketmaps/maps/europe_italy_nord-ovest-gh

000041f9 00001000 5e32870d .

000041f9 00001000 5e328150 ..

000081b0 00039453 5e328150 city_nodes.txt

000081b0 04d00064 5e328151 edges

000081b0 10d10902 5e328157 europe_italy_nord-ovest.map

000081b0 04200064 5e328158 geometry

000081b0 00b00064 5e328158 location_index

000081b0 00700064 5e328158 names

000081b0 01400064 5e328159 nodes

000081b0 00d00064 5e328159 nodes_ch_fastest_bike_node

000081b0 00d00064 5e328159 nodes_ch_fastest_car_node

000081b0 00d00064 5e32815a nodes_ch_fastest_foot_node

000081b0 00d00064 5e32815a nodes_ch_shortest_bike_node

000081b0 00d00064 5e32815a nodes_ch_shortest_car_node

000081b0 00d00064 5e32815b nodes_ch_shortest_foot_node

000081b0 00008064 5e32815b properties

000081b0 03200064 5e32815c shortcuts_fastest_bike_node

000081b0 02400064 5e32815d shortcuts_fastest_car_node

000081b0 03200064 5e32815e shortcuts_fastest_foot_node

000081b0 03d00064 5e32815f shortcuts_shortest_bike_node

000081b0 02b00064 5e328160 shortcuts_shortest_car_node

000081b0 03300064 5e328161 shortcuts_shortest_foot_node

000081b0 00000010 5e328161 version.txt

**Comment by Starcommander** [comment]
@surferjoeit 
Thanks for listing all the directories.
I have opened a new issue as the title does not match with this issue.
We will continue with the new one.

Best regards,
Paul
